mcstatus/motd.py

```diff
@@ -0,0 +1,549 @@
+MINECRAFT_COLOR_TO_HEX_JAVA[MinecraftColor.GOLD] = ("#FFAA00", "#2A2A00")
+
+MINECRAFT_COLOR_TO_ANSI = {
+    MinecraftColor.BLACK: "30",
+    MinecraftColor.DARK_BLUE: "34",
+    MinecraftColor.DARK_GREEN: "32",
+    MinecraftColor.DARK_AQUA: "36",
+    MinecraftColor.DARK_RED: "31",
+    MinecraftColor.DARK_PURPLE: "35",
+    MinecraftColor.GOLD: "33",
+    MinecraftColor.GRAY: "37",
+    MinecraftColor.DARK_GRAY: "90",
+    MinecraftColor.BLUE: "94",
+    MinecraftColor.GREEN: "92",
+    MinecraftColor.AQUA: "96",
+    MinecraftColor.RED: "91",
+    MinecraftColor.LIGHT_PURPLE: "95",
+    MinecraftColor.YELLOW: "93",
+    MinecraftColor.WHITE: "97",
+    MinecraftColor.MINECOIN_GOLD: "33",
+}
+
+
+@dataclass
+class WebColor:
+    """Raw HTML color from MOTD.
+
+    Can be found in MOTD when someone uses gradient.
+    """
+
+    hex: str
+    rgb: tuple[int, int, int]
+
+    @classmethod
+    @overload
+    def parse(cls, *, hex: str, rgb: Optional[tuple[int, int, int]] = None) -> Self:
+        ...
+
+    @classmethod
+    @overload
+    def parse(cls, *, rgb: tuple[int, int, int], hex: Optional[str] = None) -> Self:
+        ...
+
+    @classmethod
+    def parse(cls, *, hex: Optional[str] = None, rgb: Optional[tuple[int, int, int]] = None) -> Self:
+        """Parse hex or RGB color, from hex or RGB (one of them can be ``None``).
+
+        Ensures that ``WebColor.hex`` starts with '#', and generate RGB color for it,
+        if ``WebColor.rgb`` is None (default value). Or generate hex value from RGB, if
+        ``WebColor.hex`` is None (also default value).
+
+        :raises ValueError: When hex color isn't correct, so we can't generate RGB for it.
+        :raises TypeError: When hex and RGB colors is ``None``.
+        :returns: New ``WebColor`` instance.
+        """
+        if hex is None and rgb is None:
+            raise TypeError("`hex` and `rgb` parameters is None, we need one of them.")
+
+        if hex is None and rgb is not None:
+            if not all(256 > e > -1 for e in rgb):
+                raise ValueError("One of RGB colors is less than zero or more than 255: " + str(rgb))
+            hex = "#%02x%02x%02x" % rgb
+        elif hex is not None and rgb is None:
+            if not hex.startswith("#"):
+                hex = "#" + hex
+            try:
+                rgb = tuple(int(hex.lstrip("#")[i : i + 2], 16) for i in (0, 2, 4))
+            except ValueError:
+                raise ValueError(f"Incorrect hex color '{hex}', failed to generate RGB.")
+
+        return cls(hex=hex, rgb=rgb)  # type: ignore[assignment]
+
+
+@dataclass
+class TranslateString:
+    """Represents a ``translate`` field in server's answer.
+
+    This just exist, and completely ignored by our end-parsers (``ansi``, ``html`` etc. properties).
+    You can parse them with ``Motd.parsed`` attribute.
+
+    .. seealso:: `Minecraft's wiki. <https://minecraft.fandom.com/wiki/Raw_JSON_text_format#Translated_Text>`_
+    """
+
+    id: str
```

Comment:
`id` is a bit weird name for a wrapper class like this, I'd prefer `value` here. Another option would be to just inherit from string, although that might cause issues with our `isinstance` checks.

```suggestion
    value: str
```

Reply:
It actually contains an ID of the string, like `chat.type.text`. So I do not think `value` can be used here.

Reply:
What do you mean ID of the string? ID, generally stands for "Identification", how is this an ID of the string?

Reply:
Why not "chat_type"?

Reply:
> What do you mean ID of the string? ID, generally stands for "Identification", how is this an ID of the string?

Because it is Identification for string. Instead of specifying some English text, Minecraft specifies IDs and then, based on a language chosen by a user, the text gets translated.

If we refer to some guides ([like this](https://youtu.be/bbwfUbQsuGc?t=119)), it's a translation **tag**, not a string. So I renamed it [here](https://github.com/py-mine/mcstatus/compare/8b3b7ad411ed3bd1ae4af2d46cf734040ac223f3..eae1d3663789d47729e5393c8adf1417aec7c0f2).

> Why not "chat_type"?

Because `chat.type.text` is just an example, if we look to [Minecraft's wiki] (which I referenced in `seealso` tag), there is `chat.type.advancement.task` for example.

Reply:
Oh, I see, I completely misunderstood what these represented initially. In this case, the rename to `TranslationTag` is a good choice, marking this as resolved